lib/Basics/files.cpp

```diff
@@ -1671,7 +1671,7 @@ static bool CopyFileContents(int srcFD, int dstFD, ssize_t fileSize,
         break;
       }
 
-      if ((TRI_read_t)TRI_WRITE(dstFD, buf, nRead) != nRead) {
+      if (static_cast<int64_t>(TRI_WRITE(dstFD, buf, static_cast<TRI_write_t>(nRead))) != static_cast<int64_t>(nRead)) {
```

Comment:
This is actually a comment about line 1668: TRI_READ returns a signed value (ssize_t) and < 0 has a special meeing. Assigning this  to nRead which is unsigned is not good, in particular if we then compare (nRead < 1), so we miss errors! I fear we need to distinguish return types and length arguments. The usage of int64_t here might be a way out, but let's fix this rather carefully!

Reply:
will fix it

Reply:
Fixed it and added a few tests for it.

Reply:
Also created a follow-up issue, as we do not seem to use the optimized file copy method based on splice for Linux anymore: https://github.com/arangodb/planning/issues/1310